packages/kolibri-common/composables/useBaseSearch.js

```diff
@@ -161,10 +160,13 @@ export const searchKeys = [
   'grade_levels',
 ];
 
-// TBD #12517 - Will be injected in subsequent work.
-const fetchContentNodeProgress = Promise.resolve({});
-
-export default function useBaseSearch(descendant, store, router) {
+export default function useBaseSearch({
```

Comment:
Any reason for the change of function signature here?

Reply:
Hmm, imo having too many positional arguments can make the function harder to read and we need to remember the right order to send the arguments, and if we want to pass the `fetchContentNodeProgress` but not the `baseUrl` we will need to pass a null between the arguments (which imo doesnt look good). And if in the future we need to inject more dependencies to this baseSearch, we will need to add more positional arguments that can be required in some cases but optional in others. We can avoid all of these with these named arguments.

Reply:
Works for me - just wanted to make sure there was a reason :)